In [1]:
import pandas as pd
import json
import os
import requests



In [2]:
data = pd.read_csv("/Users/vickichen/Desktop/W03Project/citybikes.csv", encoding="unicode_escape")

# ========================
# preparing data
data.dropna(subset=['latitude', 'longitude'], inplace=True)
# creation of variable with lon and lat together
data['ll'] = data['latitude'].astype(str) + ',' + data['longitude'].astype(str)
data = data[data['ll'] != '0.0,0.0']
print(data.shape)
data.head()

(241, 5)


latitude   longitude  free_bikes                        stationsid  \
0  49.262487 -123.114397          19  7a19c49f486d7c0c02b3685d7b240448   
1  49.274566 -123.121817          16  32603a87cfca71d0f7dfa3513bad69d5   
2  49.279764 -123.110154          18  6d42fa40360f9a6b2bf641c7b8bb2862   
3  49.260599 -123.113504           7  66f873d641d448bd1572ab086665a458   
4  49.264215 -123.117772           7  485d4d24c803cfde829ab89699fed833   

                      ll  
0  49.262487,-123.114397  
1  49.274566,-123.121817  
2  49.279764,-123.110154  
3  49.260599,-123.113504  
4  49.264215,-123.117772

# Foursquare

Send a request to Foursquare with a small radius (1000m) for all the bike stations in your city of choice. 

In [8]:
foursquare_id = "4XHNXUWLBHWXCOTCJOAOP04HZ3GIEITQUPCVJSV25VEBGUOD"
foursquare_secret = "CLWT300A4PLYIAHUG0CK1G3JO0RW3SIO51HSBBU45EQRKLDJ"
foursquare_api = "fsq3FNZu2r3gstmBtEclZ6gZw8zcSbALLCemlKo0ReFnGSY="
fulldata=pd.DataFrame()

def get_venues(ll,id):
        
    headers = {"Accept": "application/json",
             'Authorization': foursquare_api}
    par ={"ll" :  ll,
          "radius":1000,
          "categories":str(13263)}
          #"categories":str(17114)+ "," +str(13040)}
         # "categories":str(13065) + "," + str(17114)}
    df = pd.json_normalize(results['results'])
   
    df['stationsid']=id
    global fulldata 
    fulldata= pd.concat([fulldata,df])
    return df
fsq_df = pd.DataFrame()



Parse through the response to get the POI (such as restaurants, bars, etc) details you want (ratings, name, location, etc)

Put your parsed results into a DataFrame

In [ ]:
for i, j, k,in zip(data['latitude'].values, data['longitude'].values,data['stationsid'].values):
    ll=str(round((i), 2)) + "," + str(round((j), 2))
    get_venues(ll,k)
    
    fulldata.head()
    
    
    

When i run the above codesin VSC, i got an error, says results is not defined. but my codes run fine under JupyterLab. I am not sure why it gives an error here. Please comment on this. 

Thank you!!!

# Yelp

Send a request to Yelp with a small radius (1000m) for all the bike stations in your city of choice. 

In [11]:
import pandas as pd
import json
import requests
import os



Parse through the response to get the POI (such as restaurants, bars, etc) details you want (ratings, name, location, etc)

In [12]:
data = pd.read_csv("/Users/vickichen/Desktop/W03Project/citybikes.csv", encoding="unicode_escape")

print(data.shape)
data.head()
ENDPOINT='https://api.yelp.com/v3/businesses/search'
API_KEY = "XmRtJjMG4pY29gErUXvPjWUIQyhYDEcTBIW8n64xOcsSdEg_WeO12yLLn0I7nhKiE-i1t9WUaDYEIXSgP4Z_Q-hdEDYh_MW-0AOLmPGCZgvn1uUJY0CuzJqI5fBlY3Yx"

fulldata_yelp=pd.DataFrame()

def get_venues(latitude,longitude,id):
    headers = {'Authorization': 'bearer %s' % API_KEY}
    
    PARAMETERS = {'term':'resturant',
                  'alias':'japanese',
                  'limit':50,
                  'radius':1000,
                  'latitude':latitude,
                 'longitude':longitude,
                 'rating':4.0}
    response = requests.get(url=ENDPOINT,params= PARAMETERS, headers = headers)
    
    json_response = json.loads(response.text)
    
    df = pd.json_normalize(json_response,record_path=['businesses'])
   
    df['stationsid']=id
    global fulldata_yelp 
    fulldata_yelp= pd.concat([fulldata_yelp,df])
    return df

fsq_df = pd.DataFrame()

for i, j, k,in zip(data['latitude'].values, data['longitude'].values,data['stationsid'].values):
   
    latitude=str(round((i),2))
    longitude=str(round((j),2))
    get_venues(i,j,k)
    


(241, 4)


/var/folders/fz/ntgp3y9j3p777vzmhx_1fmwr0000gn/T/ipykernel_5505/2049516725.py:28: FutureWarning: In a future version, object-dtype columns with all-bool values will not be included in reductions with bool_only=True. Explicitly cast to bool dtype instead.
  fulldata_yelp= pd.concat([fulldata_yelp,df])
/var/folders/fz/ntgp3y9j3p777vzmhx_1fmwr0000gn/T/ipykernel_5505/2049516725.py:28: FutureWarning: In a future version, object-dtype columns with all-bool values will not be included in reductions with bool_only=True. Explicitly cast to bool dtype instead.
  fulldata_yelp= pd.concat([fulldata_yelp,df])
/var/folders/fz/ntgp3y9j3p777vzmhx_1fmwr0000gn/T/ipykernel_5505/2049516725.py:28: FutureWarning: In a future version, object-dtype columns with all-bool values will not be included in reductions with bool_only=True. Explicitly cast to bool dtype instead.
  fulldata_yelp= pd.concat([fulldata_yelp,df])
/var/folders/fz/ntgp3y9j3p777vzmhx_1fmwr0000gn/T/ipykernel_5505/2049516725.py:28: FutureWarnin

Put your parsed results into a DataFrame

In [13]:
fulldata_yelp.head()

id                                 alias  \
0  Nco_OOZ4wkS8FdDGu6Mj7w               kaori-izakaya-vancouver   
1  gRepmb_vwaCDzbsjwmJ-Ig      foret-noire-patisserie-vancouver   
2  0kB2wCl1sFHf3QA7VIqxwA        lucys-eastside-diner-vancouver   
3  qO03dJ4hU0sb5I1eBSEVkA  dynasty-seafood-restaurant-vancouver   
4  u-tcnG8zdDG8QWYTctxyfg            cactus-club-cafe-vancouver   

                         name  \
0               Kaori Izakaya   
1      Foret Noire Patisserie   
2       Lucy's Eastside Diner   
3  Dynasty Seafood Restaurant   
4            Cactus Club Cafe   

                                           image_url is_closed  \
0  https://s3-media4.fl.yelpcdn.com/bphoto/sYLRql...     False   
1  https://s3-media2.fl.yelpcdn.com/bphoto/4-iXCw...     False   
2  https://s3-media3.fl.yelpcdn.com/bphoto/pzHNrq...     False   
3  https://s3-media2.fl.yelpcdn.com/bphoto/5PeLja...     False   
4  https://s3-media3.fl.yelpcdn.com/bphoto/wymPas...     False   

                                                 url  review_count  \
0  https://www.yelp.com/biz/kaori-izakaya-vancouv...          39.0   
1  https://www.yelp.com/biz/foret-noire-patisseri...          56.0   
2  https://www.yelp.com/biz/lucys-eastside-diner-...         335.0   
3  https://www.yelp.com/biz/dynasty-seafood-resta...         270.0   
4  https://www.yelp.com/biz/cactus-club-cafe-vanc...         137.0   

                                          categories  rating transactions  \
0  [{'alias': 'japanese', 'title': 'Japanese'}, {...     3.5           []   
1  [{'alias': 'desserts', 'title': 'Desserts'}, {...     4.0           []   
2  [{'alias': 'diners', 'title': 'Diners'}, {'ali...     3.0           []   
3  [{'alias': 'dimsum', 'title': 'Dim Sum'}, {'al...     3.5           []   
4  [{'alias': 'newcanadian', 'title': 'Canadian (...     3.5           []   

   ...  location.address1 location.address2  location.address3  location.city  \
0  ...   3043 Main Street              None                         Vancouver   
1  ...  236 West Broadway              None                         Vancouver   
2  ...   2708 Main Street                                           Vancouver   
3  ...     777 W Broadway           Ste 108                         Vancouver   
4  ...     575 Broadway W                                           Vancouver   

   location.zip_code location.country location.state  \
0            V5T 3G6               CA             BC   
1            V5Y 1P6               CA             BC   
2                V5T               CA             BC   
3            V5Z 4J7               CA             BC   
4            V5Z 1E6               CA             BC   

                            location.display_address price  \
0  [3043 Main Street, Vancouver, BC V5T 3G6, Canada]   NaN   
1  [236 West Broadway, Vancouver, BC V5Y 1P6, Can...    $$   
2      [2708 Main Street, Vancouver, BC V5T, Canada]    $$   
3  [777 W Broadway, Ste 108, Vancouver, BC V5Z 4J...    $$   
4    [575 Broadway W, Vancouver, BC V5Z 1E6, Canada]    $$   

                         stationsid  
0  7a19c49f486d7c0c02b3685d7b240448  
1  7a19c49f486d7c0c02b3685d7b240448  
2  7a19c49f486d7c0c02b3685d7b240448  
3  7a19c49f486d7c0c02b3685d7b240448  
4  7a19c49f486d7c0c02b3685d7b240448  

[5 rows x 25 columns]

# Comparing Results

Which API provided you with more complete data? Provide an explanation. 

The yelp API provides more information than FourSqure.

Yelp API has ratings, review_counts, image_url,locations are more easy to use.

Get the top 10 restaurants according to their rating

                        name_y  rating  \
137     Vancouver Foodie Tours     5.0   
100                 Manoush'eh     5.0   
13143                 AnnaLena     4.5   
2208         The Red Accordion     4.5   
6873                    Kingyo     4.5   
6889       Times Square Suites     4.5   
120    Le Crocodile Restaurant     4.5   
1677                     Kosoo     4.5   
112                  Sushi Jin     4.5   
111             Score on Davie     4.5 


Notes:  I did not post my code here, Its under the joiing_data notebook. See below for codes

##duplicated rest. with same locations, need to drop duplicate 
df_unique = df.drop_duplicates(subset=['name_y', 'location.display_address'], keep='first')
df_top_10 = df_unique.sort_values('rating',ascending=False).head(10)
print(df_top_10[['name_y', 'rating','location.display_address']])
